intake/cli/server/server.py

```diff
@@ -84,13 +85,25 @@ def initialize(self, cache, catalog, auth):
 
     def get(self):
         head = self.request.headers
+        page_size = self.get_argument('page[size]', None)
+        page_number = self.get_argument('page[number]', 1)
         if self.auth.allow_connect(head):
             if 'source_id' in head:
                 cat = self.cache.get(head['source_id'])
             else:
                 cat = self.catalog
             sources = []
-            for name, source in cat.walk(depth=1).items():
+            if page_size is None:
+                # Return all the results in one page. This is important for
+                # back-compat with clients that predate pagination. It may
+                # also be useful to keep things simple for clients that do not
+                # need pagination.
+                start = stop = None
+            else:
+                start = (int(page_number) - 1) * int(page_size)
+                stop = int(page_number) * int(page_size)
+            page = itertools.islice(cat.walk(depth=1).items(), start, stop)
```

Comment:
same as `iter(cat)` here instead of `walk`?

Reply:
I use the key (`name`) and the value (`source`) below. Iterating just gives me the keys. Is there a better choice than `walk`?